src/parsing/lpParser.mly

```diff
@@ -242,7 +250,13 @@ proof_end:
   | ADMITTED { make_pos $sloc Syntax.P_proof_admitted }
   | END { make_pos $sloc Syntax.P_proof_end }
 
-proof: BEGIN ts=terminated(tactic, SEMICOLON)* pe=proof_end { ts, pe }
+bracket_tactic :
+  L_CU_BRACKET tree_list=tree* R_CU_BRACKET { Tactic(None, tree_list) }
```

Comment:
For arguments, you can use short names. For instance tree_list can be replaced by l or ts.
As mentioned in another comment, the first argument of Tactic should not be an option type.
So, we should rather have something like:
```
proof: BEGIN l=subproof* pe=proof_end { l, pe }
subproof: L_CU_BRACKET l=terminated(proof_step, SEMICOLON)* R_CU_BRACKET { l }
proof_step: t=tactic l=subproof* { Tactic(t, l) }
```
Indeed, if, after begin, there are several goals, there should be a subproof for each goal. It is like if begin would be a tactic. Hence, a proof for a theorem with 2 goals should look like:
```
opaque lemma: ... :=
begin
  { ... // proof of the first goal }
  { ... // proof of the second goal }
end;
```
Note that `t1; t2;` is then simply a shorthand for `t1 { t2 };`, when t1 does not generate any additional goals but simply transforms the current goal into a new one.

Reply:
Right, the new syntax is:
```ocaml
proof:
  (*in case there are more than one goal to prove in the beginning*)
  | BEGIN l=subproof+ pe=proof_end { l, pe }
  (*in case there is just one goal to prove in the beginning*)
  | BEGIN l=subproofbis pe=proof_end { [l], pe }

subproof: L_CU_BRACKET l=terminated(proof_step, SEMICOLON)* R_CU_BRACKET { l }

subproofbis: l=terminated(proof_step, SEMICOLON)* { l }

proof_step: t=tactic l=subproof* { Tactic(t, l) }```